HelloWorld/test.java

```diff
@@ -0,0 +1,17 @@
+public class Fruit extends Food {
+  private Season ripe;
+
+  public boolean equals(Object obj) {
+    if (obj == this) {
+      return true;
+    }
+    if (obj == null) {
+      return false;
+    }
+    if (Fruit.class == obj.getClass()) { // Noncompliant; broken for child classes
+      return ripe.equals(((Fruit)obj).getRipe());
+    }
+    if (obj instanceof Fruit ) {  // Noncompliant; broken for child classes
+      return ripe.equals(((Fruit)obj).getRipe());
+    }
+    else if (obj instanceof Season) { // Noncompliant; symmetry broken for Season class
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Rename field "channel" to prevent any misunderstanding/clash with field "CHANNEL" defined on line 11. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.alumni-services-001.com/coding_rules#rule_key=squid%3AS1845)